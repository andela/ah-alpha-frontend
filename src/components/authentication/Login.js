/* eslint-disable import/no-named-as-default-member */
/* eslint-disable import/no-named-as-default */
import React from "react";
import { Modal } from "semantic-ui-react";

import LoginForm from "../containers/login/LoginForm";

const Login = () => (
  <Modal
    trigger={<div>Log in</div>}
    centered
    id="the-modal"
    style={{ display: "none" }}
    size="mini"
    dimmer
    closeIcon
  >
    <div className="modals">
      <Modal.Header>
        <h1>Login</h1>
      </Modal.Header>
      <br />
      <div>
        <LoginForm />
      </div>
    </div>
  </Modal>
);

export default Login;
