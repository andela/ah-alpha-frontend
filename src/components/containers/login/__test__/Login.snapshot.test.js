import { shallow, configure } from "enzyme";
import React from "react";
import Adapter from "enzyme-adapter-react-16";

import Login from "../../../authentication/Login";

configure({ adapter: new Adapter() });

describe("Login component being rendered", () => {
  it("should be a login component", () => {
    const tree = shallow(
      <Login />
    );
    expect(tree).toMatchSnapshot();
  });
});
