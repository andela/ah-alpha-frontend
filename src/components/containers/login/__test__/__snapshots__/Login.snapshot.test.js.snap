// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component being rendered should be a login component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "centered": true,
      "children": <div
        className="modals"
      >
        <ModalHeader>
          <h1>
            Login
          </h1>
        </ModalHeader>
        <br />
        <div>
          <Connect(LoginForm) />
        </div>
      </div>,
      "closeIcon": true,
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "id": "the-modal",
      "size": "mini",
      "style": Object {
        "display": "none",
      },
      "trigger": <div>
        Log in
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ModalHeader>
            <h1>
              Login
            </h1>
          </ModalHeader>,
          <br />,
          <div>
            <Connect(LoginForm) />
          </div>,
        ],
        "className": "modals",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h1>
              Login
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(LoginForm) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centered": true,
        "children": <div
          className="modals"
        >
          <ModalHeader>
            <h1>
              Login
            </h1>
          </ModalHeader>
          <br />
          <div>
            <Connect(LoginForm) />
          </div>
        </div>,
        "closeIcon": true,
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "id": "the-modal",
        "size": "mini",
        "style": Object {
          "display": "none",
        },
        "trigger": <div>
          Log in
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ModalHeader>
              <h1>
                Login
              </h1>
            </ModalHeader>,
            <br />,
            <div>
              <Connect(LoginForm) />
            </div>,
          ],
          "className": "modals",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <h1>
                Login
              </h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
              },
              "ref": null,
              "rendered": "Login",
              "type": "h1",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(LoginForm) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
